pkg/querier/batch/stream.go

```diff
@@ -39,64 +45,120 @@ func (bs *batchStream) atTime() int64 {
 
 func (bs *batchStream) at() (int64, float64) {
 	b := &(*bs)[0]
-	return b.Timestamps[b.Index], b.Values[b.Index]
+	return b.Timestamps[b.Index], b.SampleValues[b.Index]
```

Comment:
Should we check the batch type here, panicking with a more expressive message if it's not the right type, instead of a nil pointer panic?

Reply:
file reverted